docs/source/optim.rst

```diff
@@ -279,3 +279,30 @@ and start to collect SWA averages of the parameters at epoch 160:
 >>> torch.optim.swa_utils.update_bn(loader, swa_model)
 >>> # Use swa_model to make predictions on test data 
 >>> preds = swa_model(test_input)
+
+Functional Optimizer API
+---------------------------
+
+If you would like to update your parameters in a pure functional way, you can
```

Comment:
nit this paragraph seems to repeat itself quite a bit.
I guess the idea you want here is just that this is a functional API as it will only update the given parameters.
This can be used by users that want to just perform a single step of an optimizer for a given state.